File: rerun_cpp/src/rerun/archetypes/arrows3d_ext.cpp
#include "arrows3d.hpp"

// #define EDIT_EXTENSION

namespace rerun {
    namespace archetypes {

#ifdef EDIT_EXTENSION
        // <CODEGEN_COPY_TO_HEADER>

        /// Creates new 3D arrows pointing in the given directions, with a base at the origin (0, 0,
        /// 0).
        static Arrows3D from_vectors(Collection<components::Vector3D> vectors_) {
            return Arrows3D().with_vectors(vectors_);
        }

        // </CODEGEN_COPY_TO_HEADER>
#endif
    } // namespace archetypes
} // namespace rerun

File: crates/store/re_sdk_types/src/archetypes/arrows3d_ext.rs
use super::Arrows3D;
use crate::components::Vector3D;

impl Arrows3D {
    /// Creates new 3D arrows pointing in the given directions, with a base at the origin (0, 0, 0).
    #[inline]
    pub fn from_vectors(vectors: impl IntoIterator<Item = impl Into<Vector3D>>) -> Self {
        Self::new(vectors)
    }
}
